foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/v1_7_0/FoundationCoreJavaAiopsReferencesMigration.java

```diff
@@ -0,0 +1,93 @@
+ * #L%
+ */
+
+import com.boozallen.aissemble.upgrade.migration.AbstractAissembleMigration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.technologybrewery.baton.BatonException;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.boozallen.aissemble.upgrade.util.FileUtils.replaceLiteralInFile;
+
+public class FoundationCoreJavaAiopsReferencesMigration extends AbstractAissembleMigration {
+
+    public static final Logger logger = LoggerFactory.getLogger(FoundationCoreJavaAiopsReferencesMigration.class);
+    public static List<String> VERSION_AISSEMBLE_PRE_1_7_0 = new ArrayList<>();
+    public static List<String> VERSION_AISSEMBLE_POST_1_7_0 = new ArrayList<>();
+
+    /**
+     * Function to add packages to the VERSION_AISSEMBLE_PRE_1_7_0 & VERSION_AISSEMBLE_POST_1_7_0 list.
+     * @param n/a.
+     * @return n/a.
+     */
+    public static void addPackagesToList(){
+        VERSION_AISSEMBLE_PRE_1_7_0.add("com.boozallen.aiops.core.cdi");
+        VERSION_AISSEMBLE_PRE_1_7_0.add("com.boozallen.aiops.core.metadata");
+        VERSION_AISSEMBLE_PRE_1_7_0.add("com.boozallen.aiops.core.filestore");
+
+        VERSION_AISSEMBLE_POST_1_7_0.add("com.boozallen.aissemble.core.cdi");
+        VERSION_AISSEMBLE_POST_1_7_0.add("com.boozallen.aissemble.core.metadata");
+        VERSION_AISSEMBLE_POST_1_7_0.add("com.boozallen.aissemble.core.filestore");
+    }
+
+     /**
+     * Function to check whether the migration is necessary.
+     * @param file file to check
+     * @return shouldExecute - whether the migration is necessary.
+     */
+    @Override
+    protected boolean shouldExecuteOnFile(File file) {
+        boolean shouldExecute = false;
+        addPackagesToList();
+
+        try (BufferedReader foundationCoreJavaConfig = new BufferedReader((new FileReader(file)))) {
+
+            String line;
+            while((line = foundationCoreJavaConfig.readLine()) !=null && !shouldExecute) {
+                for (int i=0; i < VERSION_AISSEMBLE_PRE_1_7_0.size(); i++) {
+                    if (line.contains(VERSION_AISSEMBLE_PRE_1_7_0.get(i))) {
+                        return true;
+                    }
+                }
+            }
+        } catch (Exception e) {
+            logger.error("Error in determining whether the Foundation Core Java module requires a migration.");
+            throw new BatonException(e);
+        }
+        return false;
+    }
+
+    /**
+     * Performs the migration if the shouldExecuteOnFile() returns true.
+     * @param file file to migrate
+     * @return isMigrated - Whether the file was migrated successfully.
+     */
+    @Override
+    protected boolean performMigration(File file) {
+        boolean isMigrated = false;
+
+        try {
+            for (int i=0; i < VERSION_AISSEMBLE_PRE_1_7_0.size(); i++) {
```

Comment:
S: Trying to coordinate to Lists by index is very susceptible to bugs.  We should either use a `Map<String,String>` or create an class like `PackageUpdate` that has the old name and new name then use a `List<PackageUpdate>`

Reply:
Changed to a Map.